### DISCUSSION TITLE
`:events-buffer-scrollback-size` deprecated

### DISCUSSION THREAD

**Discussion by johannesnauta** [joaotavora/eglot#1447] [General]
On Emacs `28.1`, I have recently upgraded some packages after quite some time, and now I get a warning that originates from a piece of code in the [`jsonrpc`-package](https://elpa.gnu.org/packages/jsonrpc.html). This line throws a warning when `:events-buffer-scrollback-size` is used, and prompts that `:events-buffer-config` should be used instead.

```
Warning (emacs): ‘:events-buffer-scrollback-size’ deprecated. Use ‘events-buffer-config’. Disable showing Disable logging
```

The piece of code in `eglot.jl` that is calling the old is located here, after a grep search
```bash
$ grep -inr "events-buffer-scrollback-szie"
elpa/eglot-20221020.1010/eglot.el:1222:           :events-buffer-scrollback-size eglot-events-buffer-size
...
```

This already indicates that I am using eglot version `20221020.1010`, which is probably quite old. Yet, I do not know how to upgrade to the latest version and if the later version are even supported by Emacs `28.1`. Perhaps it is time to upgrade to `29+`? Or have others encountered this error perhaps?

**Comment by joaotavora** [comment]
> Yet, I do not know how to upgrade to the latest version

In that version, you should upgrade like you installed it (i.e. like you upgraded jsonrpc) , maybe with something like `M-x package-install RET eglot`.  In _later_ versions you should use `M-x eglot-upgrade-eglot`

> if the later version are even supported by Emacs 28.1

It should be.  If you notice any problems let us know.

**Comment by joaotavora** [comment]
There was some comment here about a subsequent `project.el` problem, but it was switfly deleted.  Please don't delete comments, instead just correct them in other posts.

**Reply by johannesnauta** [reply]
Ok will do. I just removed it as the problem seemingly disappeared without me providing a solution right after posting the comment. It felt therefore unnecessary to let it stay up. I apologize and will leave these up in the future.

**Reply by joaotavora** [reply]
Thanks.  Editing posts (for example for typos, like I just did) is fine.
